recipes/paho-mqtt-c/all/conanfile.py

```diff
@@ -87,21 +87,35 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.names["cmake_find_package"] = "eclipse-paho-mqtt-c"
+        self.cpp_info.names["cmake_find_package_multi"] = "eclipse-paho-mqtt-c"
+        self.cpp_info.components["_paho-mqtt-c"].names["cmake_find_package"] = self._cmake_target
+        self.cpp_info.components["_paho-mqtt-c"].names["cmake_find_package_multi"] = self._cmake_target
+        self.cpp_info.components["_paho-mqtt-c"].libs = tools.collect_libs(self)
         if self.settings.os == "Windows":
             if not self.options.shared:
-                self.cpp_info.system_libs.append("ws2_32")
+                self.cpp_info.components["_paho-mqtt-c"].system_libs.append("ws2_32")
                 if self.settings.compiler == "gcc":
-                    self.cpp_info.system_libs.extend(
+                    self.cpp_info.components["_paho-mqtt-c"].system_libs.extend(
                         ["wsock32", "uuid", "crypt32", "rpcrt4"])
         else:
             if self.settings.os == "Linux":
-                self.cpp_info.system_libs.extend(["c", "dl", "pthread"])
+                self.cpp_info.components["_paho-mqtt-c"].system_libs.extend(["c", "dl", "pthread"])
             elif self.settings.os == "FreeBSD":
-                self.cpp_info.system_libs.extend(["compat", "pthread"])
+                self.cpp_info.components["_paho-mqtt-c"].system_libs.extend(["compat", "pthread"])
             elif self.settings.os == "Android":
-                self.cpp_info.system_libs.extend(["c"])
+                self.cpp_info.components["_paho-mqtt-c"].system_libs.extend(["c"])
             else:
-                self.cpp_info.system_libs.extend(["c", "pthread"])
-        self.cpp_info.names["cmake_find_package"] = "PahoMqttC"
-        self.cpp_info.names["cmake_find_package_multi"] = "PahoMqttC"
+                self.cpp_info.components["_paho-mqtt-c"].system_libs.extend(["c", "pthread"])
+        if self.options.ssl:
+            self.cpp_info.components["_paho-mqtt-c"].requires = ["openssl::openssl"]
+
+    @property
+    def _cmake_target(self):
+        target = "paho-mqtt3"
+        target += "a" if self.options.asynchronous else "c"
+        if self.options.ssl:
+            target += "s"
+        if not self.options.shared:
+            target += "-static"
```

Comment:
poh, this is a mess :woman_dancing:

Reply:
cant we just change this upstream?

what do you think? @fpagliughi
could we harmonize the paho-mqtt-c and paho-mqtt-cpp to have sane targets and find cmake names

Reply:
I propose `PahoMqttC` and `PahoMqttCpp` for find cmake names and `paho::mqtt-c` and `paho::mqtt-cpp`
we could also replace PAHO with ECLIPSE...

Reply:
and remove all the madness with static, shared, ssl etc. changes to the name 

I mean we could also change the name based on zlib, xy, brotli etc. or use another crypto lib like mbedtls and also add this to the name -> becomes unmaintainable

Reply:
this is okay to provide different target names for different flavours of libraries.
e.g. in conan, we usually don't build static and shared in the same package. but, not everyone uses conan...
while without conan, it's pretty common practice to allow to build shared and static simultaneously (or another flavors, e.g. SSL on + SSL off, debug + release, 32 + 64, etc).
in this case, consumer has to somehow distinguish such flavours, and the best way is that he can refer to these libraries by different CMake targets - that's the most straightforward and obvious way.
so IMO there is nothing to be fixed upstream - just follow the upstream's naming in the recipe, doesn't matter how complicated it is.

Reply:
@SSE4 interesting that you support this :D

I guess you know the needed effort to support conan recipes across multiple years if everyone starts creating special options, names etc.

Reply:
Thing is he's right that not everyone uses conan and there are plenty of cases where builds generate multiple flavors and need different names.  Unfortunate but true.